### PR TITLE
Make doctor aware of configured agent sets

### LINKED ISSUE #68: [Multi-agent] Set-aware local registration, doctor, and validation
Parent epic: #60. Depends on #61 and #63. Replace the scalar local runtime record with a gitignored host map and report configured support, materialization, drift/customization, local availability, local onboarding, conformance, evidence freshness, and inert shipped adapters independently.

Validation audience is a first-class policy, not a call-site convention. Every check declares whether it applies to user/profile, maintainer/contributor, or both; its severity in each audience; the ownership class of the inspected artifact; and actionable remediation. Bare `doctor` and `doctor --runtime X` use user/profile policy. `doctor --all` and the canonical contributor validator use maintainer policy.

User/profile validation may fail for malformed or unsafe state that prevents sound operation, but it must not fail merely because an unconfigured adapter is absent, a user-owned seed is removed, optional materialization is incomplete, or evidence has harmless timezone skew. Maintainer validation enforces the complete shipped inventory and release evidence contract.

Evidence uses a deterministic 90-day freshness threshold. Stale evidence warns on user/profile surfaces and blocks maintainer release/support-claim generation. Dates up to one day ahead are tolerated as timezone skew; larger future dates warn for users and fail maintainer validation.

Acceptance:

- unavailable is distinct from unsupported, and unconfigured adapters do not fail user validation;
- each check's audience, severity, ownership class, and remediation are centrally represented and tested rather than scattered through conditionals;
- the same fixture is exercised under every applicable audience, including removed user-owned seeds, missing tool-owned files, partial optional adapters, stale evidence, one-day future skew, and implausibly future evidence;
- `doctor --runtime X` remains user-grade unless explicit maintainer mode is requested;
- existing single-runtime workflows remain compatible.

### PR DESCRIPTION
**Pull request by conorbronsdon**
## Summary

- make bare `doctor` follow tracked `contextos.workspace.json` agent intent while preserving legacy host-state fallback
- report support, component materialization, customization limits, availability, onboarding, descriptor drift, conformance declarations, and evidence freshness independently
- keep unselected adapters visible but inert, with strict explicit-runtime and maintainer scopes
- harden diagnostic path handling so malformed links, Windows reparse points, journals, and local artifacts produce structured results instead of suppressing the report
- document the set-aware scope and add regression coverage for custom paths, empty/multi-agent profiles, inert defects, evidence boundaries, and hostile local paths

Closes #68.

Stacked on #81 (`feat/neutral-root-discovery`).

## Verification

- `bash scripts/validate-all.sh`
  - 300 Python tests passed (27 intentional skips)
  - 69 portability tests passed (11 intentional skips)
  - 12 hook tests passed
  - integration, component, runtime, workspace, JSON, link, and doc-reachability checks passed
- `git diff --check`

## Exact-SHA review

Final reviewed commit: `ed1c6a230970d078aa1c96b32b76b591be847a42`

- `claude-opus-5`: no confirmed release blocker; ship
- Hermes `stealth/ox-alpha` through OpenRouter, live price $0 input / $0 output: no confirmed release blocker; pass

Three repair cycles were used. Confirmed findings fixed dangling artifact diagnostics, linked state and `.context-os` handling, Windows symlink/mount-point reparse guards, and external seed-path remapping. The repair budget is exhausted; no confirmed blocker remains.

### CODE REVIEW

File: contextos/kernel.py [hunk 1]
```diff
@@ -2447,65 +2470,487 @@ def add(name: str, status: str, detail: str) -> None:
...
+        if scoped and manifest is not None:
+            add(
+                f"runtime-onboarding:{runtime_id}",
+                "pass" if onboarding_status == "configured" else "warn",
+                onboarding_status,
+            )
+            if evidence_report["status"] in {"stale", "future", "invalid"}:
+                add(
+                    f"runtime-evidence:{runtime_id}",
+                    "warn" if evidence_report["status"] == "stale" else "fail",
```
Comment (conorbronsdon) [inline contextos/kernel.py]: **Major — a timezone-skewed "future" evidence date hard-fails bare user doctor.** `runtime_manifest(..., check_paths=False)` deliberately skips the maintainer-only future-date rejection, but doctor recomputes freshness against `effective_today = today or utc_now().date()` — and `utc_now()` is `datetime.now(timezone.utc).astimezone()`, i.e. the **local** date — then maps it via `"warn" if stale else "fail"`. A descriptor with `checked_on` equal to the author's date (valid in maintainer CI) makes bare `doctor` return overall `fail` for any user whose local calendar is still the previous day. Verified empirically: `checked_on = "2026-08-25"` with `doctor(root, today=date(2026, 8, 24))` in profile scope → overall `fail` with `runtime-evidence:hermes fail`. That fails user validation on a perfectly valid tracked artifact, against the epic's "user validation must not fail" spirit. Profile scope should warn (like `stale`) or tolerate a ±1-day skew. The future→fail path also has no test — only the stale/fresh boundary is covered.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: contextos/kernel.py [hunk 2]
```diff
@@ -2447,65 +2470,487 @@ def add(name: str, status: str, detail: str) -> None:
...
+                    else "partial"
+                )
+                component_report = {
+                    "status": component_status,
+                    "closure": closure,
+                    "missing_paths": missing_paths,
+                }
+                if scoped:
+                    materialization_status = (
+                        "fail"
```
Comment (conorbronsdon) [inline contextos/kernel.py:3279]: **Major — deleting a user-owned seed file fails bare doctor, contradicting the report's own customization stance.** The materialization presence loop requires `target.is_file()` for every non-development closure path, and a partial component is a `fail` in profile/runtime/maintainer scopes. But seed paths (`content/log.md`, `identity/who-i-am.md`, `CLAUDE.md`, `TODO.md`, …) are seeded once and thereafter user-owned per docs/component-model.md ("preserve existing `seed` paths"). Verified: with a valid `agents: ["hermes"]` profile and fully materialized closure, `rm content/log.md` yields `components:hermes fail — "partial; 1 missing; first: content/log.md"` and overall `fail` — while the same runtime report declares `customization: not-verifiable ("owned-file base hashes require the immutable bundle lock")`. Deleting or relocating one's own seeded file is exactly such unverifiable customization; seed-policy paths should at most warn (managed paths failing is defensible).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: contextos/kernel.py [hunk 3]
```diff
@@ -2447,65 +2470,487 @@ def add(name: str, status: str, detail: str) -> None:
...
+            "current"
+            if drift_status == "current"
+            else "rerun bash scripts/contextos.sh install --runtime " + drift_runtime,
+        )
+
     lock = root / ".context-os" / "apply.lock"
-    add("transaction-lock", "warn" if lock.exists() else "pass", str(lock) if lock.exists() else "none")
+    add(
+        "transaction-lock",
+        "warn" if lock.exists() else "pass",
```
Comment (conorbronsdon) [inline contextos/kernel.py]: **Minor — `transaction-lock` is the one diagnostic exempted from this PR's link hardening.** This check has no `_guard_local_state_path`/`_is_link_like` on `.context-os/apply.lock`, while the adjacent `journals`, `hosts.lock`, `proposals`, and `receipts` checks all fail closed on link-like paths. `Path.exists()` follows symlinks, so a dangling symlink at `apply.lock` reports `transaction-lock pass — none` (verified empirically), silently masking a link-like file in the local state root that every sibling check treats as a structured failure.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: contextos/kernel.py [hunk 4]
```diff
@@ -2447,65 +2470,487 @@ def add(name: str, status: str, detail: str) -> None:
...
+        )
+        drift_status = runtime_reports.get(drift_runtime, {}).get(
+            "descriptor_drift", {}
+        ).get("status", "unknown")
+        add(
+            drift_name,
+            "pass" if drift_status == "current" else "warn",
+            "current"
+            if drift_status == "current"
+            else "rerun bash scripts/contextos.sh install --runtime " + drift_runtime,
```
Comment (conorbronsdon) [inline contextos/kernel.py:3476]: **Minor — orphaned host records get actively wrong guidance, and `--all` drops them entirely.** When a `hosts.json` entry references a runtime with no shipped descriptor (legacy multi-host scope), `drift_status` is `"unknown"` and this detail tells the user to `rerun bash scripts/contextos.sh install --runtime <id>` — a command guaranteed to fail because the descriptor doesn't exist; the old code surfaced the manifest error as a `fail`. And in maintainer-all scope, `drift_targets` intersects with the registry, so an orphan host entry produces no check and no `runtimes` row at all (`test_all_scope_reports_only_shipped_registry_runtimes` blesses the omission), losing the previous stale-host-record failure signal. Suggest a dedicated `orphaned-host-record` check (warn) with `hosts.json` cleanup guidance.

Related test gap: the profile-scope drift path (configured **and** locally onboarded agent) is never exercised — all drift tests run in legacy scope or via `--runtime`, and `test_profile_scope_comes_from_tracked_agents_not_local_hosts` deliberately installs a different runtime than it configures, so the intersection is always empty.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: contextos/kernel.py [hunk 5]
```diff
@@ -2447,65 +2470,487 @@ def add(name: str, status: str, detail: str) -> None:
...
+        # malformed link that made workspace resolution fail.
+        workspace = Workspace(
+            root=root,
+            state_dir=root / DEFAULT_PATHS["state_dir"],
+            sessions_dir=root / DEFAULT_PATHS["sessions_dir"],
+            task_file=root / DEFAULT_PATHS["task_file"],
+        )
+
+    # Provider-specific instruction files belong to their runtime components,
+    # not the neutral core readiness gate.
```
Comment (conorbronsdon) [inline contextos/kernel.py:2957]: **Nit — doc drift.** This change ships the removal of the unconditional `AGENTS.md` doctor requirement (with `test_agents_instruction_is_required_only_by_selected_closure`), but docs/component-model.md still lists that removal as *pending* future work ("9. remove the remaining unconditional `AGENTS.md` doctor requirement before a Claude-only closure can omit the `agents-instructions` component"). That prerequisite is now done by this PR; the doc (untouched by this diff, unlike the three that were updated) is stale.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by conorbronsdon — COMMENTED** [review]
Reviewed the incremental diff (`feat/neutral-root-discovery..feat/set-aware-doctor`) with full head-branch context; findings were verified by running the head-branch code against reconstructed workspaces (the full kernel suite passes on head).

**Overall:** a well-structured implementation of workstream E. Scope selection (profile / legacy / runtime / maintainer-all / invalid) matches the documented precedence; inert adapters genuinely cannot add failing checks (verified: an invalid or malformed *unconfigured* descriptor yields only a structured `invalid-descriptor` row); missing local binaries and missing onboarding only warn, per the epic's "local binaries are not required merely to validate configured repository artifacts"; and the link/reparse-point hardening is real — dangling symlinks in state files, descriptors, hosts state, and artifact dirs all produce structured check results rather than crashes.

The two majors are both ways a *legitimate* user workspace still hard-fails bare `doctor` — a timezone-skewed "future" evidence date, and deletion of a user-owned seed file — which cut against the epic's core promise that user validation only fails for real repository problems. Details inline, both with reproductions. The rest is a hardening gap on `apply.lock`, wrong guidance for orphaned host records, doc drift, and test gaps.

One more untestable-guard note without an anchor: the kernel-level `doctor runtime selection and --all are mutually exclusive` raise is unreachable via the CLI (argparse group enforces it first) and untested at the API level.

---
_Generated by [Claude Code](https://claude.ai/code)_
